### PR TITLE
Fix some Sonar lint issues

### PR DESCRIPTION
**Pull request by fperot74**

### CODE REVIEW

File: api/account/api.go [hunk 1]
```diff
@@ -112,9 +112,10 @@ func ConvertToAPIAccount(userKc kc.UserRepresentation) AccountRepresentation {
 		var accreds []AccreditationRepresentation
 		for _, accredJSON := range values {
 			var accred AccreditationRepresentation
-			json.Unmarshal([]byte(accredJSON), &accred)
-			accred.Expired = keycloakb.IsDateInThePast(accred.ExpiryDate)
-			accreds = append(accreds, accred)
+			if json.Unmarshal([]byte(accredJSON), &accred) == nil {
```
Comment (harture) [inline api/account/api.go:115]: if there is an issue during unmarshalling it would be good to log it as a warning.

File: api/kyc/api.go [hunk 2]
```diff
@@ -147,9 +147,10 @@ func (u *UserRepresentation) ImportFromKeycloak(kcUser *kc.UserRepresentation) {
 		var accreds []AccreditationRepresentation
 		for _, accredJSON := range values {
 			var accred AccreditationRepresentation
-			json.Unmarshal([]byte(accredJSON), &accred)
-			accred.Expired = keycloakb.IsDateInThePast(accred.ExpiryDate)
-			accreds = append(accreds, accred)
+			if json.Unmarshal([]byte(accredJSON), &accred) == nil {
```
Comment (harture) [inline api/kyc/api.go:150]: same as before

File: api/management/api.go [hunk 3]
```diff
@@ -266,9 +267,10 @@ func ConvertToAPIUser(userKc kc.UserRepresentation) UserRepresentation {
 		var accreds []AccreditationRepresentation
 		for _, accredJSON := range values {
 			var accred AccreditationRepresentation
-			json.Unmarshal([]byte(accredJSON), &accred)
-			accred.Expired = keycloakb.IsDateInThePast(accred.ExpiryDate)
-			accreds = append(accreds, accred)
+			if json.Unmarshal([]byte(accredJSON), &accred) == nil {
```
Comment (harture) [inline api/management/api.go:270]: maybe log a warning in case of error ?

File: api/validation/api.go [hunk 4]
```diff
@@ -41,17 +41,15 @@ type CheckRepresentation struct {
...
-	prmUserGender               = "user_gender"
-	prmUserFirstName            = "user_firstName"
-	prmUserLastName             = "user_lastName"
-	prmUserEmail                = "user_emailAddress"
-	prmUserPhoneNumber          = "user_phoneNumber"
-	prmUserBirthDate            = "user_birthDate"
-	prmUserBirthLocation        = "user_birthLocation"
-	prmUserIDDocumentType       = "user_idDocType"
-	prmUserIDDocumentNumber     = "user_idDocNumber"
-	prmUserIDDocumentExpiration = "user_idDocExpiration"
```
Comment (harture) [inline api/validation/api.go:54]: why idDocExpiration is removed ?

File: cmd/keycloakb/keycloak_bridge.go [hunk 5]
```diff
@@ -62,8 +62,67 @@ var (
 	GitCommit = "unknown"
 )
 
+// RateKey is used for rate limits
```
Comment (harture) [inline cmd/keycloakb/keycloak_bridge.go:65]: J'espèere que tu as fais juste.... je suis pas bon au jeu des 7 erreurs

File: internal/constants/regexp.go [hunk 6]
```diff
@@ -2,9 +2,12 @@ package constants
 
 // Regular expressions for parameters validation
 const (
+	regExpLen255 = `^.{1,255}$`
```
Comment (harture) [inline internal/constants/regexp.go:5]: C'est vraiment pour faire plaisir a Sonar :)

File: pkg/kyc/http.go [hunk 7]
```diff
@@ -11,20 +11,23 @@ import (
...
+
+	PrmQryUserName = "username"
 )
 
 // MakeKYCHandler make an HTTP handler for the KYC endpoint.
 func MakeKYCHandler(e endpoint.Endpoint, logger log.Logger) *http_transport.Server {
-	pathParams := map[string]string{"userId": RegExpUserID}
-	queryParams := map[string]string{
-		"username": RegExpUserName,
-		"groupIds": RegExpGroupIds,
```
Comment (harture) [inline pkg/kyc/http.go:26]: Pourquoi groupsIds a disparu ?

File: api/validation/api.go (hunk 4 above)
Comment (fperot74) [reply api/validation/api.go:54]: Not used. We don't have check on this field

File: pkg/kyc/http.go (hunk 7 above)
Comment (fperot74) [reply pkg/kyc/http.go:26]: Not a parameter of the KYC API

**Pull request merged**